climada/engine/supplychain.py

```diff
@@ -0,0 +1,395 @@
+            indirect impact for each country/sector-pair.
+        total_impact (np.array): 2-dim array containing an impact-year-set with
+            total (i.e. sum direct+indirect) impact per year on each
+            country/sector-pair.
+        total_aai_agg (np.array): 1-dim np.array containing the average annual
+            total impact for each country/sector-pair.
+        io_data (dict): dictionary with four key:value-pairs:
+            coefficients (np.array): 2-dim np.array containing the technical or
+                allocation coefficient matrix, depending on employed io approach.
+            inverse (np.array): 2-dim np.array containing Leontief or Ghosh
+                inverse matrix, depending on employed io approach.
+            io_approach (str): string informing about which io approach was
+                used in calculation of indirect risk.
+            risk_structure (np.array): 3-dim np.array containing for each year
+                the risk relations between all sector/country-pairs.
+        """
+    def __init__(self):
+        """Initialization"""
+        self.mriot_data = np.array([], dtype='f')
+        self.countries = np.array([], dtype='str')
+        self.countries_iso3 = np.array([], dtype='str')
+        self.sectors = np.array([], dtype='str')
+        self.total_prod = np.array([], dtype='f')
+        self.n_countries = 0
+        self.n_sectors = 0
+        self.mriot_type = 'None'
+
+    def read_wiot(self, year=2014, file_path=SYSTEM_DIR, user_data=False,
+                  rows_range=(5,2469), col_iso3=2, cols_data_range=(4,2468),
+                  cols_sect_range=(1,61)):
+        """Read multi-regional input-output table of the WIOD project.
+        See www.wiod.org and the following paper: Timmer, M. P., Dietzenbacher,
+        E., Los, B., Stehrer, R. and de Vries, G. J. (2015), "An Illustrated
+        User Guide to the World Input–Output Database: the Case of Global
+        Automotive Production", Review of International Economics., 23: 575–605
+
+        The function supports WIOD tables release available here:
+            http://www.wiod.org/database/wiots16
+
+        Parameters:
+            year (int): Year of wiot table. Valid years go from 2000 to 2014.
+                        Default 2014.
+            file_path (str): Path to folder where the wiod table is stored.
+                            Deafult is SYSTEM_DIR. If data are not present, they
+                            will be downloaded in save_dir, i.e., ~/climada/data/results
+                            If user-defined, user_data must be set to True
+            user_data (bool): If user data are provided, Default False
+            rows_range (tuple): Start and end position of rows with data
+            col_iso3 (int): pos of col with iso3 countries
+            cols_data_range (tuple): start and end position of cols with data
+            cols_sect_range (tuple): start and end position of sectors' names
```

Comment:
since these kwargs are hardcoded and I see no reason to change them since the function is meant to be used to solely work with WIOD tables (fixed structure), I removed them and keep fix values within the function

Reply:
Either these are kwargs with default values that could be set to another value for a given reason or they are unchangeable 'hardcoded' and then are not kwargs. 

In any case, the default for kwargs must be documented in the docstrings.